syne_tune/backend/sagemaker_backend/sagemaker_backend.py

```diff
@@ -135,10 +151,23 @@ def _all_trial_results(self, trial_ids: List[int]) -> List[TrialResult]:
 
         # overrides the status return by Sagemaker as the stopping decision may not have been propagated yet.
         for trial_res in res:
-            if trial_res.trial_id in self.paused_jobs:
+            trial_id = trial_res.trial_id
```

Comment:
Well, I could not deal with self._busy_trial_ids here. I just thought since I am asking for the status anyway, I might as well use it.

Note the role of self._busy_trial_ids is really just a list of jobs for which I want to ask the status in busy_trial_ids. The shorter this list is, the better, but I don't want to miss any jobs which are busy right now.

I can remove things here, maybe this is indeed too much optimization.